homeassistant/components/growatt_server/sensor.py

```diff
@@ -62,12 +63,23 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Growatt sensor."""
-    config = config_entry.data
+    config = {**config_entry.data}
```

Comment:
For the future: I'd rewrite this part and `get_device_list` so that we don't have to copy the config entry data every time here, even if we don't update the config entry.

`get_device_list` does not need the whole config entry data. It should just receive the parameters it needs, ie plant id, username and password.